### PR TITLE
added cart and order models, added them to appcontext

### PR DESCRIPTION
**Pull request by astro8891**
@agentlewis Can you please validate these changes, thanks mate!

### CODE REVIEW

File: PantryServer/Models/Cart.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Web;
+
+
+//This will allow users to place albums in their cart without registering, but they’ll need to register to complete checkout
```
Comment (danalexilewis) [inline PantryServer/Models/Cart.cs:9]: if your going to copy paste code - at least change the references... we don't have albums in pantry - yet :)

File: PantryServer/Models/Cart.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+namespace PantryServer.Models
+{
+    public class Cart
+    {
+        // Our Cart items will have a string identifier named CartID to allow anonymous shopping
```
Comment (danalexilewis) [inline PantryServer/Models/Cart.cs]: something smells with this one. you have a product id and a product. why?
also wouldn't a cart have a collections of products. A bit like the order had a collection of order details. lets work on this on weds.

File: PantryServer/Models/OrderDetail.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿namespace PantryServer.Models
+{
+    public class OrderDetail
```
Comment (agentlewis) [inline PantryServer/Models/OrderDetail.cs]: oh i see its a list of orders. nice.

File: PantryServer/Models/Order.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        public string Address { get; set; }
+        public string City { get; set; }
+        //public string State { get; set; }
+        public string PostalCode { get; set; }
+        //public string Country { get; set; } //Maybe we want this to validate its a NZ order???
+        public string Phone { get; set; }
+        public string Email { get; set; }
+        public decimal Total { get; set; }
+        public System.DateTime OrderDate { get; set; }
+        public List<OrderDetail> OrderDetails { get; set; }
```
Comment (danalexilewis) [inline PantryServer/Models/Order.cs]: oh i see its a list of orders. nice.

File: PantryServer/Models/OrderDetail.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    {
+        public int OrderDetailId { get; set; }
+        public int OrderId { get; set; }
+        public int ProductId { get; set; }
+        public int Quantity { get; set; }
+        public decimal UnitPrice { get; set; }
```
Comment (danalexilewis) [inline PantryServer/Models/OrderDetail.cs]: price is in the product. no...?

File: PantryServer/Models/OrderDetail.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+        public virtual Product product { get; set; }
+        public virtual Shop shop { get; set; }
```
Comment (danalexilewis) [inline PantryServer/Models/OrderDetail.cs]: not sure if we need to reference shop - we have access to it from the product object.

**Comment by danalexilewis** [comment]
Also aware that we need to be able to place an order against multiple shops. Though for mvp lets do it for one at a time :) but good to remember.

**Comment by astro8891** [comment]
Hey mate thanks for the feedback.
I need a spike on DB thinking, obviously missing something crucial with the thought patterns.
Will catch up on Wednesday - no seed method will block our way this time!

Date: Sun, 22 May 2016 14:05:07 -0700

**Pull request merged**